drivers/i2s/i2s_cavs.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ * @brief I2S bus (SSP) driver for Intel CAVS.
+ *
+ * Limitations:
+ * - DMA is used in simple single block transfer mode (with linked list
+ *   enabled) and "interrupt on full transfer completion" mode.
+ */
+
+#ifndef _I2S_CAVS_H_
+#define _I2S_CAVS_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct ssp {
+	u32_t ssc0;		/* 0x00 - Control0 */
+	u32_t ssc1;		/* 0x04 - Control1 */
+	u32_t sss;		/* 0x08 - Status */
+	u32_t ssit;		/* 0x0C - Interrupt Test */
+	u32_t ssd;		/* 0x10 - Data */
+	u32_t reserved0[5];
+	u32_t ssto;		/* 0x28 - Time Out */
+	u32_t sspsp;		/* 0x2C - Programmable Serial Protocol */
+	u32_t sstsa;		/* 0x30 - TX Time Slot Active */
+	u32_t ssrsa;		/* 0x34 - RX Time Slot Active */
+	u32_t sstss;		/* 0x38 - Time Slot Status */
+	u32_t reserved1;
+	u32_t ssc2;		/* 0x40 - Command / Status 2 */
+	u32_t sspsp2;		/* 0x44 - Programmable Serial Protocol 2 */
+	u32_t ssc3;		/* 0x48 - Command / Status 3 */
+	u32_t ssioc;		/* 0x4C - IO Control */
+};
+
```

Comment:
More broadly: why is this stuff in a header at all?  It defines internal data that is only used by or included in one other file, and exposes no APIs for use by other parts of the system.  Just put this in the C file.

Reply:
This is more of a personal choice, i feel. I wanted to keep some things in the header file to keep the size of the source file reasonable. Also, i wasn't sure if the things that i put in the header file would not be used anywhere else (except the corresponding .c). That said, i can merge the .c an .h files if you insist.